bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -234,64 +238,63 @@ public String getIdentifier()
 
   public void flush(final int writeOffset)
   {
-    //logger.debug("size = {}, processingOffset = {}, nextOffset = {}, writeOffset = {}", size, processingOffset,
-    //    nextOffset.integer, writeOffset);
-    flush:
-    do {
-      while (size == 0) {
-        size = VarInt.read(last.data, processingOffset, writeOffset, nextOffset);
-        if (nextOffset.integer > -5 && nextOffset.integer < 1) {
-          if (writeOffset == last.data.length) {
-            nextOffset.integer = 0;
-            processingOffset = 0;
-            size = 0;
-          }
-          break flush;
-        } else if (nextOffset.integer == -5) {
-          throw new RuntimeException("problemo!");
-        }
+    size = VarInt.read(last.data, processingOffset, writeOffset, nextOffset);
```

Comment:
If I remove this block one unit test is failing. So keeping it as is, will analyze the reason for the unit test failure.